File: SPCServer/springboot/src/main/java/com/lcaohoanq/shoppe/models/Token.java
package com.lcaohoanq.shoppe.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "tokens")
public class Token {

    @Id
    @SequenceGenerator(name = "tokens_seq", sequenceName = "tokens_id_seq", allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "tokens_seq")
    @Column(name="id", unique=true, nullable=false)
    @JsonProperty("id")
    private Long id;

    @Column(name = "token", length = 255)
    private String token;

    @Column(name = "refresh_token", length = 255)
    private String refreshToken;

    @Column(name = "token_type", length = 50)
    private String tokenType;

    @Column(name = "expiration_date")
    private LocalDateTime expirationDate;

    @Column(name = "refresh_expiration_date")
    private LocalDateTime refreshExpirationDate;

    @Column(name = "is_mobile")
    private boolean isMobile;

    private boolean revoked;
    private boolean expired;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

}

File: kt-service/src/main/kotlin/com/lcaohoanq/ktservice/entities/Token.kt
package com.lcaohoanq.ktservice.entities

import BaseEntity
import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.persistence.*
import java.time.LocalDateTime
import java.util.*

@Entity
@Table(name = "token")
class Token(

    @Id
    @Column(name = "id", unique = true, nullable = false)
    @JsonProperty("id")
    val id: UUID? = UUID.randomUUID(),
    var token: String,
    var refreshToken : String,
    var tokenType: String,
    var expirationDate: LocalDateTime,
    var refreshExpirationDate: LocalDateTime,
    var isMobile: Boolean,
    var revoked: Boolean,
    var expired: Boolean,

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    val user: User

): BaseEntity() {
}